src/main/java/sirius/biz/storage/layer2/BlobDuplicator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.storage.layer2;
+
+import sirius.db.KeyGenerator;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.health.Exceptions;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Defines parameters used to duplicate a {@link Blob} in a new temporary copy.
+ */
+public class BlobDuplicator {
+
+    @Part
+    private static BlobStorage blobStorage;
+
+    @Part
+    private static KeyGenerator keyGenerator;
+
+    private final Blob blob;
+    private String storageSpace;
+    private final Set<String> variants = new HashSet<>();
+
+    /**
+     * Creates a new duplicator for the given blob.
+     *
+     * @param blob the blob to duplicate
+     * @return a new duplicator for the given blob
+     */
+    public static BlobDuplicator create(Blob blob) {
+        return new BlobDuplicator(blob);
+    }
+
+    private BlobDuplicator(Blob blob) {
+        this.blob = blob;
+        this.storageSpace = blob.getSpaceName();
+    }
+
+    /**
+     * Adds a variant name to be duplicated among with the blob.
+     *
+     * @param variantName the variant name to duplicate
+     * @return the duplicator itself for fluent method calls
+     */
+    public BlobDuplicator withVariant(String variantName) {
+        this.variants.add(variantName);
```

Comment:
What if the variant does not exist?

Reply:
It will silently not be copied.
I choose this way because one could define variants in DB but not necessarily be attached to a real "converter".
At the same time, a real variant name might not exist for the blob being copied.